shared/common-adapters/icon.desktop.js

```diff
@@ -47,39 +47,57 @@ class Icon extends Component<Exact<Props>, void> {
       hoverColor = 'inherit'
     } else {
       color =
-        (this.props.style && this.props.style.color) ||
-        color ||
-        (this.props.opacity ? globalColors.lightGrey : globalColors.black_40)
+        this.props.color || color || (this.props.opacity ? globalColors.lightGrey : globalColors.black_40)
       hoverColor =
-        (this.props.style && this.props.style.hoverColor) ||
+        this.props.hoverColor ||
         hoverColor ||
         (this.props.opacity ? globalColors.black : globalColors.black_75)
     }
 
     const isFontIcon = iconType.startsWith('iconfont-')
-    const fontSizeHint = shared.fontSize(iconType)
+    const fontSizeHint = this.props.fontSize ? {fontSize: this.props.fontSize} : shared.fontSize(iconType)
     const onClick = this.props.onClick
       ? e => {
           e.stopPropagation()
           this.props.onClick && this.props.onClick(e)
         }
       : null
 
-    if (isFontIcon) {
-      const cleanStyle = {
-        fontFamily: 'kb',
-        speak: 'none',
-        fontStyle: 'normal',
-        fontWeight: 'normal',
-        fontVariant: 'normal',
-        textTransform: 'none',
-        lineHeight: 1, // NOT 1px, just 1
-        WebkitFontSmoothing: 'antialiased',
-        ...this.props.style,
-      }
-      // We have to blow these styles away else FontIcon gets confused and will overwrite what it calculates
-      delete cleanStyle.color
-      delete cleanStyle.hoverColor
+    const hasContainer = (this.props.onClick && this.props.style) || isFontIcon
+
+    const imgStyle = collapseStyles([
+      desktopStyles.noSelect,
+      !hasContainer ? this.props.style : {},
+      this.props.iconStyle,
+      onClick ? desktopStyles.clickable : {},
+      this.props.color ? {color: color} : {},
+    ])
+
+    const iconElement = (
```

Comment:
we should make this a tertiary so we don't create if it ifs not and just use this value in the render, like
```
const iconElement = isFontIcon 
  ?  String.fromCharCode(iconMeta[iconType].charCode || 0)
  : <img
        className={this.props.className}
        draggable="false"
        title={this.props.hint}
        style={imgStyle}
        onClick={onClick}
        srcSet={iconTypeToSrcSet(iconType)}
      />

...
<Box...>
{iconElement}
```